### PR TITLE
* Remove onaudioprocess event handler before processor has been deleted.

### PR DESCRIPTION
**Pull request by a8568730**
因為原底若kā錄音擋起來，console會跳錯誤 "TypeError: recBuffers.push is undefined"。
原專案 PR [#37](https://github.com/higuma/web-audio-recorder-js/pull/37#issuecomment-704311448) 有人講是 race condition。

### CODE REVIEW

File: lib/WebAudioRecorder.js [hunk 1]
```diff
@@ -126,6 +127,7 @@
       if (this.isRecording()) {
         this.input.disconnect();
         this.processor.disconnect();
+	this.processor.onaudioprocess = null;
```
Comment (sih4sing5hong5) [inline lib/WebAudioRecorder.js]: 排版？

File: lib/WebAudioRecorder.js [hunk 2]
```diff
@@ -116,6 +116,7 @@
       if (this.isRecording()) {
         this.input.disconnect();
         this.processor.disconnect();
+        this.processor.onaudioprocess = null;
```
Comment (sih4sing5hong5) [inline lib/WebAudioRecorder.js:119]: 這影響著sannh？

**Review by sih4sing5hong5 — APPROVED** [review]

**Review by sih4sing5hong5 — APPROVED** [review]

File: lib/WebAudioRecorder.js (hunk 2 above)
Comment (a8568730) [reply lib/WebAudioRecorder.js:119]: 我認為是為著 https://github.com/i3thuan5/web-audio-recorder-js/blob/master/lib/WebAudioRecorder.js#L98-L102 onaudioprocess 內底有送信號 `worker.postMessage({ command: "record", buffer: buffer });`，叫後壁錄音。

**Comment by a8568730** [comment]
另外發現歹消息，套件內底 [createScriptProcessor](https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/createScriptProcessor) beh pha-sé--ah。
> This feature was replaced by [AudioWorklets](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorklet and the [AudioWorkletNode](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorkletNode) interface.

File: lib/WebAudioRecorder.js (hunk 1 above)
Comment (a8568730) [reply lib/WebAudioRecorder.js]: [9e1bf30](https://github.com/i3thuan5/web-audio-recorder-js/pull/1/commits/9e1bf3057eab9f5a3908d71f1f424fe0495179b3)

**Pull request merged**